homeassistant/components/screenlogic/services.py

```diff
@@ -27,44 +32,100 @@
+            vol.Optional(ATTR_CONFIG_ENTRY): selector.ConfigEntrySelector(
+                {
+                    "integration": DOMAIN,
+                }
+            ),
+            **cv.ENTITY_SERVICE_FIELDS,
+            vol.Required(ATTR_COLOR_MODE): vol.In(SUPPORTED_COLOR_MODES),
+        }
+    ),
+    cv.has_at_least_one_key(ATTR_CONFIG_ENTRY, *cv.ENTITY_SERVICE_FIELDS),
 )
 
-TURN_ON_SUPER_CHLOR_SCHEMA = cv.make_entity_service_schema(
+TURN_ON_SUPER_CHLOR_SCHEMA = BASE_SERVICE_SCHEMA.extend(
     {
-        vol.Optional(ATTR_RUNTIME, default=24): vol.Clamp(
-            min=MIN_RUNTIME, max=MAX_RUNTIME
+        vol.Optional(ATTR_RUNTIME, default=24): vol.All(
+            vol.Coerce(int), vol.Clamp(min=MIN_RUNTIME, max=MAX_RUNTIME)
         ),
     }
 )
 
 
 @callback
-def async_load_screenlogic_services(hass: HomeAssistant, entry: ConfigEntry):
+def async_load_screenlogic_services(hass: HomeAssistant):
     """Set up services for the ScreenLogic integration."""
 
     async def extract_screenlogic_config_entry_ids(service_call: ServiceCall):
         if not (
-            screenlogic_entry_ids := [
-                entry_id
-                for entry_id in await async_extract_config_entry_ids(hass, service_call)
-                if (entry := hass.config_entries.async_get_entry(entry_id))
-                and entry.domain == DOMAIN
-            ]
+            screenlogic_entry_ids := await async_extract_config_entry_ids(
+                hass, service_call
+            )
         ):
-            raise HomeAssistantError(
+            raise ServiceValidationError(
                 f"Failed to call service '{service_call.service}'. Config entry for"
                 " target not found"
             )
         return screenlogic_entry_ids
 
+    async def get_coordinators(
+        service_call: ServiceCall,
+    ) -> list[ScreenlogicDataUpdateCoordinator]:
+        entry_ids: set[str]
+        if entry_id := service_call.data.get(ATTR_CONFIG_ENTRY):
+            entry_ids = {entry_id}
+        else:
+            ir.async_create_issue(
+                hass,
+                DOMAIN,
+                "service_target_deprecation",
+                breaks_in_ha_version="2024.8.0",
+                is_fixable=True,
+                is_persistent=True,
+                severity=ir.IssueSeverity.WARNING,
+                translation_key="service_target_deprecation",
+            )
+            entry_ids = await extract_screenlogic_config_entry_ids(service_call)
+
+        coordinators: list[ScreenlogicDataUpdateCoordinator] = []
+        for entry_id in entry_ids:
+            config_entry: ConfigEntry | None = hass.config_entries.async_get_entry(
+                entry_id
+            )
+            if not config_entry:
+                raise ServiceValidationError(
+                    f"Failed to call service '{service_call.service}'. Config entry '{entry_id}' not found"
+                )
+            if not config_entry.domain == DOMAIN:
```

Comment:
I think the config entry selector in the service schema already validates the config entry domain.

Reply:
Through the UI yes, but it appears that raw service calls (script or automation) with a valid entry_id make it through to that check even if they are not of the proper domain.  Should I not be worrying about that use-case?

Reply:
Yeah, looks like we don't implement any validation from the selector config in the backend.

Ok, let's keep this for now.